### ISSUE TITLE
Add utility to compute bounding box of resulting prediction

### ISSUE THREAD

**Issue by adamjstewart** [microsoft/torchgeo#35]
### Rationale

For GeoDatasets, at sampling time, we should know the CRS, bounding box, and resolution of the image that gets sampled. As the image is passed through transforms like `Resample` or `Warp`, we should be able to recompute the new CRS/bbox/res pretty easily. However, depending on the padding and stride used in convolutional and pooling layers, the bounding box and resolution may change significantly as the image is passed through the model. In order to save or stitch together our predictions, we'll need to be able to compute the new bbox/res.

### Description

We have two possible options:

1. Build a utility that takes in a `nn.Module` (the neural network) and computes the resulting bbox/res.
2. Modify PyTorch's builtin modules (at least the convolution and pooling ones) to take the bbox/res as input and modify them directly.

In the short-term, we will likely go with 1 since it involves the least work. In the long-term, we may end up going with 2 since we'll want to be able to design networks that can take advantage of this kind of geospatial information.
